File: csharp/ql/test/query-tests/Security Features/CWE-089-2/SqlInjection.cs
using System;
using Microsoft.Data;
using Microsoft.Data.SqlClient;

namespace Test
{
    class SqlInjection
    {
        string connectionString;
        System.Windows.Forms.TextBox box1;

        public void MakeSqlCommand()
        {
            // BAD: Text from a local textbox
            using (var connection = new SqlConnection(connectionString))
            {
                var queryString = "SELECT ITEM,PRICE FROM PRODUCT WHERE ITEM_CATEGORY='"
                  + box1.Text + "' ORDER BY PRICE"; // $ Source[cs/sql-injection]
                var cmd = new SqlCommand(queryString); // $ Alert[cs/sql-injection]
                var adapter = new SqlDataAdapter(cmd); // $ Alert[cs/sql-injection]
            }

            // BAD: Input from the command line.
            using (var connection = new SqlConnection(connectionString))
            {
                var queryString = "SELECT ITEM,PRICE FROM PRODUCT WHERE ITEM_CATEGORY='"
                  + Console.ReadLine() + "' ORDER BY PRICE"; // $ Source[cs/sql-injection]
                var cmd = new SqlCommand(queryString); // $ Alert[cs/sql-injection]
                var adapter = new SqlDataAdapter(cmd); // $ Alert[cs/sql-injection]
            }
        }
    }
}

File: go/ql/test/query-tests/Security/CWE-089/SqlInjection.go
package main

import (
	"database/sql"
	"fmt"
	"net/http"
)

func handler(db *sql.DB, req *http.Request) {
	q := fmt.Sprintf("SELECT ITEM,PRICE FROM PRODUCT WHERE ITEM_CATEGORY='%s' ORDER BY PRICE", // $ Source[go/sql-injection]
		req.URL.Query()["category"])
	db.Query(q) // $ Alert[go/sql-injection]
}
